components/pkg-export-tar/plan.sh

```diff
@@ -0,0 +1,87 @@
+pkg_name=hab-pkg-export-tar
+_pkg_distname=$pkg_name
+pkg_origin=core
+pkg_version=$(cat "$PLAN_CONTEXT/../../VERSION")
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_license=('Apache-2.0')
+pkg_deps=(core/coreutils core/findutils core/gawk core/grep core/bash core/tar core/gzip core/hab)
+pkg_build_deps=(
+  core/musl core/zlib-musl core/xz-musl core/bzip2-musl core/libarchive-musl
+  core/openssl-musl core/libsodium-musl
+  core/coreutils core/rust core/gcc core/make
+)
+
+pkg_bin_dirs=(bin)
+
+bin=$_pkg_distname
+
+_common_prepare() {
+  do_default_prepare
+
+  # Can be either `--release` or `--debug` to determine cargo build strategy
+  build_type="--release"
+  build_line "Building artifacts with \`${build_type#--}' mode"
+
+  # Used by the `build.rs` program to set the version of the binaries
+  export PLAN_VERSION="${pkg_version}/${pkg_release}"
+  build_line "Setting PLAN_VERSION=$PLAN_VERSION"
+
+  if [ -z "$HAB_CARGO_TARGET_DIR" ]; then
+    # Used by Cargo to use a pristine, isolated directory for all compilation
+    export CARGO_TARGET_DIR="$HAB_CACHE_SRC_PATH/$pkg_dirname"
+  else
+    export CARGO_TARGET_DIR="$HAB_CARGO_TARGET_DIR"
+  fi
+  build_line "Setting CARGO_TARGET_DIR=$CARGO_TARGET_DIR"
+}
+
+do_prepare() {
```

Comment:
Most of this code is shared across all of the Rust exporters. I think it might be a good opportunity to abstract it. It would probably look pretty similar to https://github.com/habitat-sh/builder/blob/master/support/ci/builder-base-plan.sh

Reply:
Good point!  Due to the time pressure of what we need to do for the on-prem builder in the next couple of months, I'd like to delay doing this abstraction for now.  I have created an issue around abstracting this logic here https://github.com/habitat-sh/habitat/issues/4522

Reply:
Indeed! I stumbled across this myself, resulting in https://github.com/habitat-sh/habitat/pull/4524

@nellshamrell I agree with you 100% on pushing the abstraction off a bit... I was tempted to do it myself, but we have bigger fish to fry at the moment.